assembl/views/auth/views.py

```diff
@@ -165,8 +195,12 @@ def login_view(request):
     if request.scheme == "http"\
             and asbool(config.get("accept_secure_connection")):
         raise HTTPFound("https://" + request.host + request.path_qs)
-    return get_login_context(
-        request, request.matched_route.name == 'login_forceproviders')
+    force_providers = request.matched_route.name.endswith('_forceproviders')
+    if request.matched_route.name == 'contextual_login':
+        contextual_login = get_social_autologin(request)
+        if contextual_login:
+            raise HTTPFound(contextual_login)
```

Comment:
If there is a `contextual_login, redirect to that request? Otherwise, no exception thrown, continue as usual?